clients/cobol-lsp-vscode-extension/src/services/SmartTabSettings.ts

```diff
@@ -0,0 +1,53 @@
+import * as vscode from "vscode";
+import { SETTINGS_TAB_CONFIG } from "../constants";
+
+export class TabRule {
+  // tslint:disable-next-line:no-unnecessary-initializer
+  public constructor(
+    public stops: number[],
+    public maxPosition: number,
+    public regex: string | undefined = undefined,
+  ) {}
+}
+
+export class TabSettings {
+  public constructor(public rules: TabRule[], public defaultRule: TabRule) {}
+}
+
+/**
+ * Retrieves and parse tab settings configuration that can be boolean, array or an object
+ * @returns a TabSettings object
+ */
+export function getTabSettings(): TabSettings {
+  const config = vscode.workspace.getConfiguration().get(SETTINGS_TAB_CONFIG);
+  let settings = new TabSettings([], new TabRule([0, 6, 7, 11], 72));
+  if (Array.isArray(config)) {
+    const stops = config as number[];
```

Comment:
add type checks in a follow-up PR?

Reply:
I added types and input validation for the configuration 👍